service/core.go

```diff
@@ -63,20 +65,49 @@ type SetContainerer interface {
 	SetContainer(Host)
 }
 
+type metricsCore struct {
+	scope           tally.RootScope
+	runtimeReporter *metrics.RuntimeReporter
+}
+
+func (mc *metricsCore) Metrics() tally.Scope {
+	return mc.scope
+}
+
+func (mc *metricsCore) RuntimeMetrics() *metrics.RuntimeReporter {
+	return mc.runtimeReporter
+}
+
+type tracerCore struct {
+	tracer       opentracing.Tracer
+	tracerCloser io.Closer
+	tracerConfig jaegerconfig.Configuration
+}
+
+func (tc *tracerCore) Tracer() opentracing.Tracer {
+	return tc.tracer
+}
+
+type loggingCore struct {
```

Comment:
This was originally from Aiden's comment about the struct getting bloated. Since it was a result of adding a new field to this struct, I think it's ok to have it as part of this diff. 
Good idea Anup, don't want to add all that refactor as part of this though. Will make it part of a new diff.